apps/meteor/client/views/room/composer/messageBox/MessageBox.tsx

```diff
@@ -117,7 +122,60 @@ const MessageBox = ({
 	const composerPlaceholder = useMessageBoxPlaceholder(t('Message'), room);
 
 	const [typing, setTyping] = useReducer(reducer, false);
+	const [uploadfiles, setUploadfiles] = useState<File[]>([]);
+	const dispatchToastMessage = useToastMessageDispatch();
+	const maxFileSize = useSetting('FileUpload_MaxFileSize') as number;
+	function handlefileUpload(fileslist: File[], resetFileInput?: () => void) {
+		setUploadfiles((prevFiles) => [...prevFiles, ...fileslist]);
 
+		resetFileInput?.();
+	}
+
+	const handleremove = (index: number) => {
+		if (!uploadfiles) {
+			return;
+		}
+		const temp = uploadfiles.filter((_, i) => {
```

Comment:
If you already have the index, you can use `splice` to remove the item from the array https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice

Reply:
yes sir added splice method